libminifi/include/utils/ValueParser.h

```diff
@@ -0,0 +1,193 @@
+    return *this;
+  }
+
+  template<typename Out>
+  ValueParser& parseLongLong(Out& out) {
+    static_assert(is_non_narrowing_convertible<long long, Out>::value, "Expected lossless conversion from long long");  // NOLINT
+    long long result;  // NOLINT
+    auto len = safeCallConverter(std::strtoll, result);
+    if ( len == 0 ) {
+      throw ParseException("Couldn't parse long long");
+    }
+    offset += len;
+    out = {result};
+    return *this;
+  }
+
+  template<typename Out>
+  ValueParser& parseUInt32(Out& out) {
+    static_assert(is_non_narrowing_convertible<uint32_t, Out>::value, "Expected lossless conversion from uint32_t");
+    parseSpace();
+    if (offset < str.length() && str[offset] == '-') {
+      throw ParseException("Not an unsigned long");
+    }
+    unsigned long result;  // NOLINT
+    auto len = safeCallConverter(std::strtoul, result);
+    if ( len == 0 ) {
+      throw ParseException("Couldn't parse uint32_t");
+    }
+    if (result > (std::numeric_limits<uint32_t>::max)()) {
+      throw ParseException("Cannot convert unsigned long to uint32_t");
+    }
+    offset += len;
+    out = {static_cast<uint32_t>(result)};
+    return *this;
+  }
+
+  template<typename Out>
+  ValueParser& parseUnsignedLongLong(Out& out) {
+    static_assert(is_non_narrowing_convertible<unsigned long long, Out>::value, "Expected lossless conversion from unsigned long long");  // NOLINT
+    parseSpace();
+    if (offset < str.length() && str[offset] == '-') {
+      throw ParseException("Not an unsigned long");
+    }
+    unsigned long long result;  // NOLINT
+    auto len = safeCallConverter(std::strtoull, result);
+    if ( len == 0 ) {
+      throw ParseException("Couldn't parse unsigned long long");
+    }
+    offset += len;
+    out = {result};
+    return *this;
+  }
+
+  template<typename Out>
+  ValueParser& parseBool(Out& out) {
+    parseSpace();
+    if (std::strncmp(str.c_str() + offset, "false", std::strlen("false")) == 0) {
+      offset += std::strlen("false");
+      out = false;
+    } else if (std::strncmp(str.c_str() + offset, "true", std::strlen("true")) == 0) {
+      offset += std::strlen("true");
+      out = true;
+    } else {
+      throw ParseException("Couldn't parse bool");
+    }
+    return *this;
+  }
+
+  void parseEnd() {
+    parseSpace();
+    if (offset < str.length()) {
+      throw ParseException("Expected to parse till the end");
+    }
+  }
+
+ private:
+  /**
+   *
+   * @tparam T
+   * @param converter
+   * @param out
+   * @return the number of characters used during conversion, 0 for error
+   */
+  template<typename T>
+  std::size_t safeCallConverter(T (*converter)(const char* begin, char** end, int base), T& out) {
+    const char* const begin = str.c_str() + offset;
+    char* end;
+    errno = 0;
+    T result = converter(begin, &end, 10);
+    if (end == begin || errno == ERANGE) {
+      return 0;
+    }
+    out = result;
+    return end - begin;
+  }
+
+  void parseSpace() {
```

Comment:
very minor, but I would call this `skipWhitespace()`

Reply:
done